### PR TITLE
TST: Insert 'match' to bare pytest raises

### PR DESCRIPTION
**Pull request by ShaharNaveh**
- [x] ref #30999
- [ ] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/test_errors.py
Comment (simonjayhawkins) [inline pandas/tests/test_errors.py]: the match argument is treated as a regex, did you mean "^$", i.e. no message?

File: pandas/tests/test_take.py
Comment (simonjayhawkins) [inline pandas/tests/test_take.py]:
```suggestion
            "indices are out-of-bounds"
```

**Review by simonjayhawkins — COMMENTED** [review]
Thanks @MomIsBestFriend for the PR.

File: pandas/tests/test_errors.py
Comment (MomIsBestFriend) [reply pandas/tests/test_errors.py]: Correct, will fix.

(Thank you)

**Comment by MomIsBestFriend** [comment]
@simonjayhawkins Do you mind if I open a master issue, instead of #23922 ?
There's a lot of noise there about how to find a bare pytest raises, I have a test case and I can point at the line number at each file with a bare pytest raise, and will make a list of occurrences (similar to #29886)

---

Do you think it's a good idea?

**Comment by simonjayhawkins** [comment]
> Do you think it's a good idea?

The format of #29886 worked well for community engagement. so OK with opening a new issue to supersede #23922.

**Review by simonjayhawkins — APPROVED** [review]
Thanks @MomIsBestFriend lgtm pending green.

File: pandas/tests/test_lib.py
Comment (WillAyd) [inline pandas/tests/test_lib.py]: minor but could simplify with a group or even character class

```suggestion
            "index 100 is out of bounds for axis (0|1) with size 100|"
```

**Review by WillAyd — CHANGES_REQUESTED** [review]

File: pandas/tests/test_downstream.py
Comment (WillAyd) [inline pandas/tests/test_downstream.py]:
```suggestion
        "returned non-zero exit status 1."
```

I don't think need the raw prefix here either

**Comment by jbrockmendel** [comment]
pending addressing comments from @WillAyd, LGTM

File: pandas/tests/test_lib.py
Comment (MomIsBestFriend) [reply pandas/tests/test_lib.py]: That's really smart!

**Comment by simonjayhawkins** [comment]
Thanks @MomIsBestFriend

**Pull request merged**
